XenAdmin/TabPages/NICPage.cs

```diff
@@ -216,13 +217,32 @@ public PIFRow(PIF pif)
                     device = PIFMetrics.device_name;
                     busPath = PIFMetrics.pci_bus_path;
                 }
-                for (int i = 0; i < 9; i++)
+                for (int i = 0; i < 10; i++)
                 {
                     Cells.Add(new DataGridViewTextBoxCell());
                     updateCell(i);
                 }
 
-            }
+                DelegatedAsyncAction action = new DelegatedAsyncAction(pif.Connection,

```

Comment:
The action is only needed in the case where we call Network_sriov.get_remaining_capacity. In all the other cases (!pif.SriovCapable(), !pif.IsSriovPhysicalPIF()), the Cell value can be assigned immediatelly.
With this in mind, the function updateCell() seems more suitable for this code.